sdk/identity/identity/src/credentials/azureCliCredential.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { createSpan } from "../util/tracing";
+import { AuthenticationErrorName } from "../client/errors";
+import { CanonicalCode } from "@opentelemetry/types";
+import { logger } from "../util/logging";
+
+import * as child_process from "child_process";
+
+export interface CredentialClient {
```

Comment:
Since this is specific to the `AzureCliCredential` I'd call this `AzureCliCredentialClient` and call the implementation `DefaultAzureCliCredentialClient`.  Since `CredentialClient` is being exposed on the public interface of this library, it'll be confusing to see something with a more general name than the usage pattern implies.

Reply:
This also needs some basic documentation to make api-extractor happy.

Reply:
Fixed with the refactor